orchagent/portsorch.cpp

```diff
@@ -1040,22 +1072,40 @@ bool PortsOrch::setPortAdminStatus(Port &port, bool state)
     attr.id = SAI_PORT_ATTR_ADMIN_STATE;
     attr.value.booldata = state;
 
+    /* Update the host_tx_ready to false before setting admin_state, when admin state is false */
+    if (!state) 
+    {
+        m_portStateTable.hset(port.m_alias, "host_tx_ready", "false");
+        SWSS_LOG_INFO("Set admin status DOWN host_tx_ready to false to port pid:%" PRIx64,
+                port.m_port_id);
+    }
+    
     sai_status_t status = sai_port_api->set_port_attribute(port.m_port_id, &attr);
     if (status != SAI_STATUS_SUCCESS)
     {
         SWSS_LOG_ERROR("Failed to set admin status %s to port pid:%" PRIx64,
                        state ? "UP" : "DOWN", port.m_port_id);
+        m_portStateTable.hset(port.m_alias, "host_tx_ready", "false");
         task_process_status handle_status = handleSaiSetStatus(SAI_API_PORT, status);
         if (handle_status != task_success)
         {
             return parseHandleSaiStatusFailure(handle_status);
         }
     }
 
-    SWSS_LOG_INFO("Set admin status %s to port pid:%" PRIx64,
-                    state ? "UP" : "DOWN", port.m_port_id);
-
-    setGearboxPortsAttr(port, SAI_PORT_ATTR_ADMIN_STATE, &state);
+    bool gbstatus = setGearboxPortsAttr(port, SAI_PORT_ATTR_ADMIN_STATE, &state);
+    if (gbstatus != true)
```

Comment:
host_tx_ready will be set to false two times when admin_status is down. Can you fix that? May be do this:

```
if (state) {
     if ((gbstatus == true) && (status == SAI_STATUS_SUCCESS)) {
         m_portStateTable.hset(port.m_alias, "host_tx_ready", "true");
     } else {
         m_portStateTable.hset(port.m_alias, "host_tx_ready", "false");
     }
}
```

Reply:
@prgeor , No. host_tx_ready will be set to false only once during admin down. In case of admin down, it is best to disable the device on the line side first (optics), so setting the host_tx_ready to false before calling SAI to disable it in NPU/PHY.

If there is a SAI API failure during admin down/up config, then host_tx_ready will be set to false.